src/script/engines/lua.c

```diff
@@ -340,19 +415,47 @@ struct mScriptEngineContext* _luaCreate(struct mScriptEngine2* engine, struct mS
 		mLOG(SCRIPT, ERROR, "Error in dostring while initializing sockets: %s\n", lua_tostring(luaContext->lua, -1));
 		lua_pop(luaContext->lua, 1);
 	} else {
+		struct mScriptValue* errors = mScriptValueAlloc(mSCRIPT_TYPE_MS_TABLE);
 		int i;
 		lua_getglobal(luaContext->lua, "socket");
 		lua_getfield(luaContext->lua, -1, "ERRORS");
 		for (i = 0; i < _mScriptSocketNumErrors; i++) {
 			const struct _mScriptSocketError* err = &_mScriptSocketErrors[i];
 			if (err->message) {
 				lua_pushstring(luaContext->lua, err->message);
+				struct mScriptValue* key = mScriptValueAlloc(mSCRIPT_TYPE_MS_S32);
+				key->value.s32 = err->err;
+				struct mScriptValue* message = mScriptStringCreateFromASCII(err->message);
+				mScriptTableInsert(errors, key, message);
+				mScriptValueDeref(key);
+				mScriptValueDeref(message);
 			} else {
 				lua_pushnil(luaContext->lua);
 			}
 			lua_seti(luaContext->lua, -2, err->err);
 		}
 		lua_pop(luaContext->lua, 2);
+
+		mScriptEngineExportDocNamespace(&luaContext->d, "socket", (struct mScriptKVPair[]) {
+			mSCRIPT_KV_PAIR(ERRORS, errors),
+			mSCRIPT_KV_PAIR(tcp, mSCRIPT_VALUE_DOC_FUNCTION(socket_tcp)),
+			mSCRIPT_KV_PAIR(bind, mSCRIPT_VALUE_DOC_FUNCTION(socket_bind)),
+			mSCRIPT_KV_PAIR(connect, mSCRIPT_VALUE_DOC_FUNCTION(socket_connect)),
+			mSCRIPT_KV_SENTINEL
+		});
+		mScriptValueDeref(errors);
+		mScriptEngineSetDocstring(&luaContext->d, "socket", "A basic TCP socket library");
+		mScriptEngineSetDocstring(&luaContext->d, "socket.ERRORS",
+			"Error strings corresponding to the C.SOCKERR error codes, indexed both by name and by value");
```

Comment:
Is there a way to link this to the enum documentation?

Reply:
Which, `socket.ERRORS` or `C.SOCKERR`? The latter will be linked by this convention already, but the former I couldn't get working properly without breaking other stuff.

Reply:
`C.SOCKERR`. If it works automatically then great.